2_process/src/create_group_task_plan.R

```diff
@@ -0,0 +1,60 @@
+create_group_task_makefile <- function(makefile, task_plan, remake_file, final_targets) {
+  scipiper::create_task_makefile(
+    makefile=makefile, task_plan=task_plan,
+    include=remake_file,
+    packages=c("purrr", "dplyr", "readr", "feather"),
+    sources=c("2_process/src/munge_group_files.R"),
+    finalize_funs = "gather_lake_info",
+    final_targets = final_targets
+  )
+}
+
+create_group_tasks <- function(task_ids, log_folder){
+  
+  # prepare a data.frame with one row per task
+  tasks <- data_frame(task_id=task_ids[1:2]) %>%
+    mutate(task_name = sprintf("group_%s", substr(task_id, 1,2)))
+  
+  get_lake_ids <- scipiper::create_task_step(
+    step_name = 'get_lake_ids',
+    target_name = function(task_name, step_name, ...){
+      sprintf("%s_lake_ids", task_name)
+    },
+    command = function(task_name, ...){
+      cur_task <- dplyr::filter(rename(tasks, tn=task_name), tn==task_name)
+      psprintf("get_lakes_from_group(", 
+               "sb_groups = sb_groups,",
+               "grp_id = I('%s'))" = cur_task$task_id
+      )
+    } 
+  )
+  
+  combine_zipped_files <- scipiper::create_task_step(
+    step_name = 'combine_zipped_files',
+    target_name = function(task_name, step_name, ...){
+      sprintf("2_process/log/merged_data_%s.rds", task_name)
+    },
+    command = function(task_name, ...){
+      cur_task <- dplyr::filter(rename(tasks, tn=task_name), tn==task_name)
+      psprintf("unzip_and_merge_files(", 
+               "target_name = target_name,",
+               "lake_ids = %s_lake_ids," = task_name,
+               "irradiance_zipfile = I('1_fetch/out/irradiance_%s.zip')," = cur_task$task_id,
```

Comment:
Other than the `fn_out_template`, these shouldn't have I() blocks, right? There isn't a reason to hide changes to the zip file from remake as far as I can tell, and it seems risky to do so.

Reply:
well, those zip files aren't targets the way it is written now. There is a target for all irradiance downloaded but the files are by group. But I think that will change once I implement the sc_indicate method?